src/http-client.js

```diff
@@ -0,0 +1,218 @@
+        return {
+            ...options,
+            baseUrl,
+            method,
+            expBackoffMillis,
+            expBackoffMaxRepeats,
+            retryOnStatusCodes,
+        };
+    }
+
+    _getAxiosConfig(options) {
+        return {
+            baseURL: `${options.baseUrl}${options.basePath}`,
+            url: options.url,
+            method: options.method,
+            headers: {
+                'user-agent': CLIENT_USER_AGENT,
+                'accept-encoding': 'gzip',
+                'content-type': 'application/json; charset=utf-8',
+                ...options.headers,
+            },
+            params: {
+                ...options.qs,
+                token: options.token,
+            },
+            data: options.body,
+            // Override the default transform function to get unparsed response.
+            transformResponse: res => res,
+            httpAgent: this.httpAgent,
+            httpsAgent: this.httpsAgent,
+            validateStatus: null,
+            responseType: options.encoding === null ? 'arraybuffer' : null,
+        };
+    }
+
+    _getResponseLike(axiosResponse) {
+        let response;
+
+        if (axiosResponse.request.res) {
+            // It is in node
+            // return standard Node.Js response-like object;
+            response = axiosResponse.request.res;
+            response.body = axiosResponse.data;
+        } else {
+            const { data, status, statusText, headers, config } = axiosResponse;
+            response = {
+                body: data,
+                statusCode: status,
+                statusText,
+                headers,
+                config,
+            };
+        }
+        return response;
+    }
+
+    async _call(options) {
+        const {
+            expBackoffMillis,
+            expBackoffMaxRepeats,
+            retryOnStatusCodes,
+        } = options;
+
+        this.stats.calls++;
+        let iteration = 0;
+
+        const axiosConfig = this._getAxiosConfig(options);
+
+        const makeRequest = async (bail) => {
+            iteration += 1;
+            let statusCode;
+            let response;
+            let error;
+
+            try {
+                this.stats.requests++;
+                const axiosResponse = await this.axiosInstance.request(axiosConfig); // eslint-disable-line
+                // TODO Emulate Request API for now
+                response = this._getResponseLike(axiosResponse);
+                statusCode = response ? response.statusCode : null;
+
+                // TODO Retest this with Axios
+                // It may happen that response is incomplete but request package silently returns original
+                // response as string instead of throwing an error. So we call JSON.parse() manually here.
+                // If parsing throws then the request gets retried with exponential backoff.
+                if (options.json && response.body) response.body = JSON.parse(response.body);
```

Comment:
Shouldn't we parse JSON this only when status code is < 300. 

Also keep in mind that API can redirect (for example reads from key-value store might be redirected to S3)

Reply:
I'm quite certain `axios` handles redirects before it even gives you access to the response.

Reply:
Ok. But we are still parsing 4xx and 5xx response although it gets retried below.

Reply:
Yeah. On the other hand, we use the `response.body` down on line 188. The whole piece of code is from the original client implementation with some very minor fixes to make it `axios` compatible. It needs another refactoring, but let's keep it out of this PR so that we can release a working browser build first and deal with the optimizations later.